lib/puppet/indirector/json.rb

```diff
@@ -52,7 +52,7 @@ def load_json_from_file(file, key)
     json = nil
 
     begin
-      json = File.read(file)
+      json = File.read(file).force_encoding(Encoding::UTF_8)
```

Comment:
Going through the call I suggested here at the beginning is synonymous with File.read, so you can still call force_encoding on it.

https://github.com/puppetlabs/puppet/blob/stable/lib/puppet/file_system.rb#L125